src/main/java/rx/plugins/RxJavaPlugins.java

```diff
@@ -149,7 +164,27 @@ private static Object getPluginImplementationViaProperty(Class<?> pluginClass) {
          * This will only happen during system startup thus it's okay to use the synchronized
          * System.getProperties as it will never get called in normal operations.
          */
-        String implementingClass = System.getProperty("rxjava.plugin." + classSimpleName + ".implementation");
+        
+        String defaultKey = "rxjava.plugin." + classSimpleName + ".implementation";
+        String implementingClass = props.getProperty(defaultKey);
+        
+        for (Map.Entry<Object, Object> e : props.entrySet()) {
+            String key = e.getKey().toString();
+            if (key.startsWith("rxjava.plugin.") && key.endsWith(".class")) {
+                String value = e.getValue().toString();
+                
+                if (classSimpleName.equals(value)) {
+                    String index = key.substring(0, key.length() - 6).substring(14);
+                    
+                    String implKey = "rxjava.plugin." + index + ".impl";
+                    
+                    implementingClass = props.getProperty(implKey);
```

Comment:
Initialization is the best time to crash the app because it has almost no state at this point of time and developer could detect problem immediately.

> I'm sure the developer who specified the override will notice his override doesn't work.

But how he will notice if usually developers override schedulers or error handlers? :)

Fail fast, fail early!